customerio.go

```diff
@@ -207,3 +208,48 @@ func (c *CustomerIO) request(method, url string, body interface{}) error {
 
 	return nil
 }
+
+const (
+	// Identifier Types
+	IdentifierTypeID    = "id"
+	IdentifierTypeEmail = "email"
+	IdentifierTypeCioID = "cio_id"
+)
+
+// MergeCustomers sends a request to Customer.io to merge two customer profiles together.
+func (c *CustomerIO) MergeCustomers(primaryIDType, primaryID, secondaryIDType, secondaryID string) error {
+	if !isValidIDType(primaryIDType) {
+		return ParamError{Param: "primaryIDType"}
+	}
+	if strings.TrimSpace(primaryID) == "" {
+		return ParamError{Param: "primaryID"}
+	}
+
+	if !isValidIDType(secondaryIDType) {
+		return ParamError{Param: "secondaryIDType"}
+	}
+	if strings.TrimSpace(secondaryID) == "" {
+		return ParamError{Param: "secondaryID"}
+	}
+
+	return c.request("POST",
+		fmt.Sprintf("%s/api/v1/merge_customers", c.URL),
+		map[string]interface{}{
+			"primary": map[string]string{
```

Comment:
then we could do a custom JSON serializer for `Identifier` that serializes it into `{"email": "value"}` or whatever